app/src/main/java/com/vanced/manager/ui/fragments/HomeFragment.kt

```diff
@@ -56,6 +61,20 @@ class HomeFragment : BindingFragment<FragmentHomeBinding>() {
             homeRefresh.setOnRefreshListener { viewModel.fetchData() }
             isFetching.observe(viewLifecycleOwner) { homeRefresh.isRefreshing = it }
 
+            val prefs = PreferenceManager.getDefaultSharedPreferences(context)
+            if (prefs.contains("LastVersionCode")) {
```

Comment:
Well I removed the `contains` statement but I think this was not a so good idea. I divided this *problem* in 4 different cases that may happen :
1. App was updated
1.1 Value is in SharedPrefs -> the dialog will pop up and value in SharedPrefs will be upated
1.2 Value is not in SharedPrefs -> the statement in the if block evaluates to false so the else block saves VERSION_CODE in SharedPrefs
2. App was not updated
2.1 Value is in SharedPrefs -> however the if statement will evaluate to false since it is a less and not a less or equal so the else block will be run and the value in SharedPrefs be written again although it is useless since the value is already up-to-date
2.2 Value is not in SharedPrefs -> the statement in the if block evaluates to false so the else block saves VERSION_CODE in SharedPrefs

So as you can see there is a problem for the 2.1 situation (which is also the situation that is likely to happen most of the time). 
Using the `contains` statement ensured that no useless SharedPrefs edit is made

Reply:
Hmm, I guess you're right with that.

Reply:
Well if I keep the first version of my code, changelog will not be shown on first time install because the first if statement with `contains` will be false to the only thing that will happen is that VERSION_CODE will be saved to SharedPrefs.

> Edit: it's 2021 and Gboard still doesn't know what Changelog is.

Hmm ? I didn't understand but it's clear that Gboard is weird lol

Reply:
> Well if I keep the first version of my code, changelog will not be shown on first time install because the first if statement with `contains` will be false to the only thing that will happen is that VERSION_CODE will be saved to SharedPrefs.

Then I guess it's good to be merged

Reply:
So should I change everything back to the first version ? If yes, I would also change the defaultValue back to -1 since using the contains function ensures that defaultValue will never be thrown. What do u think ?

Reply:
> So should I change everything back to the first version ? If yes, I would also change the defaultValue back to -1 since using the contains function ensures that defaultValue will never be thrown. What do u think ?

Seems like a good idea to me

Reply:
done ;)